receiver/otlpreceiver/otlp.go

```diff
@@ -94,32 +129,41 @@ func (r *otlpReceiver) Start(_ context.Context, host component.Host) error {
 	var err error
 	r.startServerOnce.Do(func() {
```

Comment:
can we move all this code to a "startInternal" instead of having everything in this anonymous func?

Reply:
Done.